assets/aml-benchmark/tests/test_dataset_preprocessor.py

```
@@ -0,0 +1,184 @@
+                None, None
+            ),
+            ("quac_org", INPUT_DATASET, None, os.path.join(CUSTOM_SCRIPT_PATH, "quac_textgen_babel.py"), None),
+            (
+                "mnli", INPUT_DATASET,
+                """ {"premise":{{premise}}, "hypothesis":{{hypothesis}},"label":{{label|string}}} """,
+                None, '{"column_name":"label", "0":"NEUTRAL", "1":"ENTAILMENT", "2":"CONTRADICTION"}'
+            ),
+        ],
+    )
+    def test_dataset_preprocessor_as_component(
+        self,
+        dataset_name: str,
+        dataset: str,
+        template_input: str,
+        script_path: str,
+        encoder_config: str,
+    ) -> None:
+        """Dataset Preprocessor component test."""
+        with open(
+            os.path.join(os.path.dirname(INPUT_DATASET), "process_one_example.jsonl"), "w"
+        ) as writer:
+            with open(INPUT_DATASET, "r") as reader:
+                for line in reader:
+                    out_row = json.loads(line)
+                    if out_row.get('name') == dataset_name:
+                        del out_row['name']
+                        writer.write(json.dumps(out_row) + "\n")
+        dataset = os.path.join(os.path.dirname(INPUT_DATASET), "process_one_example.jsonl")
+        ml_client = get_mlclient()
+        exp_name = f"{self.EXP_NAME}-{dataset_name}"
+        pipeline_job = self._get_pipeline_job(
+            dataset,
+            template_input,
+            script_path,
+            encoder_config,
+            "Dataset pre-processor pipeline test",
+            pipeline_file="dataset_preprocessor_pipeline.yaml",
+        )
+        # submit the pipeline job
+        try:
+            pipeline_job = ml_client.create_or_update(
+                pipeline_job, experiment_name=exp_name
+            )
+            ml_client.jobs.stream(pipeline_job.name)
+            print(pipeline_job)
+        except Exception as e:
+            print(e)
+            print('Failed with exception')

```

Can we please remove this? Let it fail if there is a failure, it's easier to investigate that way.

---

Done. Thanks.